zerver/webhooks/github/view.py

```diff
@@ -302,16 +318,75 @@ def get_push_commits_body(helper: Helper) -> str:
+            "action": lambda: self.payload["action"].tame(check_string),
+            "optional_title": lambda: f' {self.template_values["title"]()}'
+            if self.include_title
+            else "",
+            "category": lambda: self.payload["discussion"]["category"]["name"].tame(check_string),
+            "title": lambda: self.payload["discussion"]["title"].tame(check_string),
+            "body": lambda: self.payload["discussion"]["body"].tame(check_string),
+            "repository_name": lambda: self.payload["repository"]["name"].tame(check_string),
+            "new_repository_name": lambda: self.payload["changes"]["new_repository"]["name"].tame(
+                check_string
+            ),
+            "discussion_number": lambda: self.payload["discussion"]["number"].tame(check_int),
+            "new_discussion_number": lambda: self.payload["changes"]["new_discussion"][
+                "number"
+            ].tame(check_int),
+            "label": lambda: self.payload["label"]["name"].tame(check_string),
+            "old_category": lambda: self.payload["changes"]["category"]["from"]["name"].tame(
+                check_string
+            ),
+            "locked_reason": lambda: f' as {self.payload["discussion"]["active_lock_reason"].tame(check_string)}'
+            if self.payload["discussion"]["active_lock_reason"]
+            else "",
+            "closed_reason": lambda: self.payload["discussion"]["state_reason"].tame(check_string),
+            "answer_field": lambda: "old_answer"
+            if self.payload["action"].tame(check_string) == "unanswered"
+            else "answer",
+            "answer_url": lambda: self.payload[self.template_values["answer_field"]()][
+                "html_url"
+            ].tame(check_string),
+            "answer_body": lambda: self.payload[self.template_values["answer_field"]()][
+                "body"
+            ].tame(check_string),
+            "comment_id": lambda: self.payload[self.template_values["answer_field"]()]["id"].tame(
+                check_int
+            ),
+        }
+
+    @override
+    def __getitem__(self, key: str) -> str | int:
+        return self.template_values[key]()
+
+
 def get_discussion_body(helper: Helper) -> str:
     payload = helper.payload
-    return DISCUSSION_TEMPLATE.format(
-        author=get_sender_name(payload),
-        url=payload["discussion"]["html_url"].tame(check_string),
-        body=payload["discussion"]["body"].tame(check_string),
-        category=payload["discussion"]["category"]["name"].tame(check_string),
-        discussion_id=payload["discussion"]["number"].tame(check_int),
-        title=payload["discussion"]["title"].tame(check_string),
-    )
+    action = get_discussion_action(payload)
+    DISCUSSION_TEMPLATE = DISCUSSION_TEMPLATES[action]
+    context = LazyContext(payload, helper.include_title)
+    return DISCUSSION_TEMPLATE.format_map(context)
```

Comment:
How do discussion events show up in the set of "event type" options for filtering in the settings UI? I'm not sure if it'll just be a single "discussion" item or one per each of these dozen subtypes.

Reply:
> How do discussion events show up in the set of "event type" options for filtering in the settings UI?

There's just 1 "event" `discussion` and all these "actions" within that event. So, just 1 entry.
While 16 entries just for the discussion event would pollute the list, not being able to opt out of particular actions is inconvenient indeed.
Would it be worth extending event filtering to include one or more sub-event layers?